src/scratch.h

```diff
@@ -0,0 +1,35 @@
+/**********************************************************************
+ * Copyright (c) 2017 Andrew Poelstra	                              *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+
+#ifndef _SECP256K1_SCRATCH_
+#define _SECP256K1_SCRATCH_
+#include <sys/types.h>
+
+/* The typedef is used internally; the struct name is used in the public API
+ * (where it is exposed as a different typedef) */
+typedef struct secp256k1_scratch_space_struct {
+    void *data;
+    size_t offset;
+    size_t init_size;
+    size_t max_size;
```

Comment:
Can we add an `error_callback` pointer to this structure and remove it from the arguments of `secp256k1_scratch_resize`? It would avoid the need to pass error callbacks into a bunch of higher level functions.